tests/src/test/java/org/eclipse/hono/tests/http/HttpTestBase.java

```diff
@@ -911,6 +911,36 @@ public void testUploadMessageFailsForUnauthorizedGateway(final VertxTestContext
             .onComplete(ctx.succeedingThenComplete());
     }
 
+    /**
+     * Verifies that the HTTP adapter rejects messages with content length > max body limit size.
+     *
+     * @param ctx The test context
+     */
+    @Test
+    @Timeout(timeUnit = TimeUnit.SECONDS, value = 20)
+    public void testUploadMessageFailsForBodyLimitSize(final VertxTestContext ctx) {
```

Comment:
I think it would be better to put the addition of this test method in a separate PR, since it's related to the bodyHandler changes ( #3277) and not the idle timeout changes made in this PR.